packages/mantine/src/form/use-form.ts

```diff
@@ -0,0 +1,31 @@
+import {useForm as useMantineForm} from '@mantine/form';
+import {_TransformValues, UseFormInput, GetInputProps} from '@mantine/form/lib/types';
+
+export const useForm = <
+    Values = Record<string, unknown>,
+    TransformValues extends _TransformValues<Values> = (values: Values) => Values
+>(
+    options: UseFormInput<Values, TransformValues> = {}
+) => {
+    const form = useMantineForm(options);
+
+    const getInputProps: GetInputProps<Values> = (
+        path,
+        {type = 'input', withError = type === 'input', withFocus = true} = {}
+    ) => {
+        const originalPayload = form.getInputProps(path, {type, withError, withFocus});
+        if (Array.isArray(originalPayload.value)) {
```

Comment:
This can be dangerous because all form components that have an array as their value will automatically be overloaded.

Reply:
Yes they will be automatically overloaded. this is right. 

But in the end this should not change anything. Because if the component is an original one from Mantine. It will most certainly leverage the already implemented`onChange` method that has never been removed or changed.

Since the useForm getInputProps only passes new props to the component, nothing should be changed in actual implementations.